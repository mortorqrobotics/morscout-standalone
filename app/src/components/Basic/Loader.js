import React from "react";
import Loader from "react-loader-spinner";

export default function Loading() {
  return <Loader type="Bars" />;
}
