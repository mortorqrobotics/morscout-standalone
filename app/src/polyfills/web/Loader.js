import React from 'react';
import Loader from 'react-loader';

export default function Loading() {
  return (<Loader type="spin" />);
}
