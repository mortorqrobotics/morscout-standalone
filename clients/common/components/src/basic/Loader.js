import React from "react";
import { View, StyleSheet } from "react-native";
import Loader from "react-loader-spinner";

const style = StyleSheet.create({
  Centered: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center"
  }
});

export default function Loading() {
  return (
    <View style={style.Centered}>
      <Loader type="Bars" />
    </View>
  );
}
