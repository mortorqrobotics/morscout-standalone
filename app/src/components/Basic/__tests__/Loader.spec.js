// __tests__/Loader.spec.js
import React from "react";
import renderer from "react-test-renderer";
import Loader from "../Loader";

test("Loader renders correctly", async () => {
  expect.assertions(1);
  const tree = renderer.create(React.createElement(Loader, {}, [])).toJSON();
  expect(tree).toMatchSnapshot();
});
