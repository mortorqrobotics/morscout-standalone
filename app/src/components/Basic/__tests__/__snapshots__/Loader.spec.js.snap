// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
<div>
  <svg
    fill="#71238"
    height={80}
    viewBox="0 0 135 140"
    width={80}
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="120"
      rx="6"
      width="15"
      y="10"
    >
      <animate
        attributeName="height"
        begin="0.5s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="120;110;100;90;80;70;60;50;40;140;120"
      />
      <animate
        attributeName="y"
        begin="0.5s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="10;15;20;25;30;35;40;45;50;0;10"
      />
    </rect>
    <rect
      height="120"
      rx="6"
      width="15"
      x="30"
      y="10"
    >
      <animate
        attributeName="height"
        begin="0.25s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="120;110;100;90;80;70;60;50;40;140;120"
      />
      <animate
        attributeName="y"
        begin="0.25s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="10;15;20;25;30;35;40;45;50;0;10"
      />
    </rect>
    <rect
      height="140"
      rx="6"
      width="15"
      x="60"
    >
      <animate
        attributeName="height"
        begin="0s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="120;110;100;90;80;70;60;50;40;140;120"
      />
      <animate
        attributeName="y"
        begin="0s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="10;15;20;25;30;35;40;45;50;0;10"
      />
    </rect>
    <rect
      height="120"
      rx="6"
      width="15"
      x="90"
      y="10"
    >
      <animate
        attributeName="height"
        begin="0.25s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="120;110;100;90;80;70;60;50;40;140;120"
      />
      <animate
        attributeName="y"
        begin="0.25s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="10;15;20;25;30;35;40;45;50;0;10"
      />
    </rect>
    <rect
      height="120"
      rx="6"
      width="15"
      x="120"
      y="10"
    >
      <animate
        attributeName="height"
        begin="0.5s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="120;110;100;90;80;70;60;50;40;140;120"
      />
      <animate
        attributeName="y"
        begin="0.5s"
        calcMode="linear"
        dur="1s"
        repeatCount="indefinite"
        values="10;15;20;25;30;35;40;45;50;0;10"
      />
    </rect>
  </svg>
</div>
`;
